docs/src/consensus/transaction.md

```diff
@@ -0,0 +1,176 @@
+
+ - `RemovalRecordsIntegrity :: (transaction_kernel_mast_hash : Digest) ⟶ (inputs_salted_utxos_hash : Digest)` Establishes that all removal records (which themselves are commitments to input UTXOs) are correctly computed and applicable. Specifically:
+   - divine the input UTXOs
+   - divine the salt
+   - divine the mutator set accumulator and authenticate it against the given transaction kernel MAST hash
+   - for each input UTXO:
+     - compute the removal record index set
+     - verify that at least one index has not been seen before
+   - hash the list of removal records and authenticate it against the given transaction kernel MAST hash
+   - output the hash of the salted UTXOs.
+ - `KernelToOutputs :: (transaction_kernel_mast_hash : Digest) ⟶ (outputs_salted_utxos_hash : Digest)` Collects the output UTXOs into a more digestible format. Specifically:
+   - divine the output UTXOs
+   - divine the salt
+   - for each output UTXO:
+     - divine the commitment randomness
+     - compute the canonical commitment
+   - hash the list of canonical commitments
+   - authenticate the list of canonical commitments against the given transaction kernel MAST hash
+   - output the hash of the salted UTXOs.
+ - `CollectLockScripts :: (inputs_salted_utxos_hash : Digest) ⟶ (lock_script_hashes : [Digest])` Collects the lock script hashes into a list. Specifically:
+   - divine the input UTXOs
+   - divine the salt
+   - authenticate the salted UTXOs against the given hash digest
+   - for each UTXO:
+     - collect the lock script hash
+   - output all lock script hashes.
+ - `LockScript :: (transaction_kernel_mast_hash : Digest) ⟶ ∅` Unlocks a single input UTXO. The concrete program value of a lockscript depends on the UTXO. By default, this program is created by a generation address, in which case it asserts knowledge of a preimage to a hardcoded digest.
+ - `CollectTypeScripts :: (inputs_salted_utxos_hash : Digest) × (outputs_salted_utxos_hash : Digest) ⟶ (type_script_hashes : [Digest])` Collects the type scripts into a more digestible format. Specifically:
+   - divine all input UTXOs
+   - divine the salt for input UTXOs
+   - authenticate the salted input UTXOs against the given hash digest
+   - divine all output UTXOs
+   - divine the salt for output UTXOs
+   - authenticate the salted output UTXOs against the given hash digest
+   - for each input or output UTXO:
+     - collect the type script hash
+   - filter out duplicates
+   - output the unique type script hashes
+ - `TypeScript :: (transaction_kernel_mast_hash : Digest) ⟶ ∅` Authenticates the correct evolution of all UTXOs of a given type. The concrete program value depends on the token types involved in the transaction. For Neptune's native currency, Neptune Coins, the type script asserts that *a)* all output amounts are positive, and *b)* the sum of all input amounts is greater than or equal to the fee plus the sum of all output amounts.
```

Comment:
Do we need a `TypeScript` execution proof for *each* digest returned by `CollectTypeScripts`? That doesn't seem clearly specified.

Reply:
Yes. Thanks; will clarify.